packages/bindgen/src/templates/node-wrapper.ts

```diff
@@ -130,20 +143,25 @@ export function generate({ spec: rawSpec, file }: TemplateContext): void {
     injectables.push(cls.jsName);
 
     let body = "";
+    let body_browser = "";
 
     // It will always be accessed via this name rather than a static to enable optimizations
     // that depend on the symbol not changing.
     const symb = `_${cls.rootBase().jsName}_Symbol`;
 
     if (!cls.base) {
       // Only root classes get symbols and constructors
-      both(`const ${symb} = Symbol("Realm.${cls.jsName}.external_pointer");`);
+      all(`const ${symb} = Symbol("Realm.${cls.jsName}.external_pointer");`);
+      browserLines.push(
+        `const _${cls.jsName}_registery = new FinalizationRegistry(nativeModule.${cls.jsName}_deleter);`,
+      );
```

Comment:
I think you need to do the finalization registry for every class, not just base classes. It will be a bit tricky though because you need to only do the registration in the most-derived class's constructor. I suggest making the constructors look something like this:

```js
constructor(ptr, calledFromSubclass) {
  ${cls.base ? 'super(ptr, true)', `this[${symb}] = ptr;`}
  if (!calledFromSubclass) _${cls.jsName}_registery.register(this, ptr);
}
```

I doubt that the extra branch matters much, but if we want to avoid it where we know it isn't needed, we can do `${cls.subclasses.length == 0 ? '' : 'if (!calledFromSubClass)} _${cls.jsName}_registery.register(this, ptr);`. Might give us a slight size savings, and isn't too much extra complication, so might be worth it.